### ISSUE TITLE
[graphql-code-generator] 0.18.0 does not generate client specific types

### ISSUE THREAD

**Issue by FuNn1esT** [dotansimha/graphql-code-generator#1425]
**Describe the bug**

graphql-code-generator 0.18.0 will not generate client-specific types from `ts` file

**To Reproduce**

1. Use `schema.gql` as schema

```gql
# schema.gql
schema {
  query: Query
}

type Query {
  me: User
  user(id: ID!): User
  allUsers: [User]
}

enum Role {
  USER
  ADMIN
}

type User {
  id: ID!
  username: String!
  email: String!
  role: Role!
}
```

2. Use `test.gql` and `test.ts` as documents

```gql
# test.gql
query findUser($userId: ID!) {
  user(id: $userId) {
    ...UserFields
  }
}

fragment UserFields on User {
  id
  username
  role
}
```

```ts
// test.ts
import { gql } from 'apollo-boost';

export const testQuest = gql`
  query findUser($userId: ID!) {
    user(id: $userId) {
      ...UserFields
    }
  }

  fragment UserFields on User {
    id
    username
    role
  }
`;
```

3. Use `codegen.yml` as config

```yml
# codegen.yml
overwrite: true
schema: 'schema.gql'
generates:
  types-from-gql.ts:
    documents: 'test.gql'
    config:
      noNamespaces: true
    plugins:
      - typescript-common
      - typescript-client
  types-from-ts.ts:
    documents: 'test.ts'
    config:
      noNamespaces: true
    plugins:
      - typescript-common
      - typescript-client
```

4. Run the following command: `gql-gen --config codegen.yml`

**Expected behavior**

generated file `types-from-ts.ts` should be the same as `types-from-gql.ts`

when the graphql-code-generator version is 0.17.0 it works well, but when the version is 0.18.0 the `types-from-ts.ts` does not generate client-specific types

`package.json` file

```diff
{
  "scripts": {
    "codegen": "gql-gen --config codegen.yml"
  },
  "devDependencies": {
-   "graphql-code-generator": "^0.17.0",
+   "graphql-code-generator": "^0.18.0",
    "graphql-codegen-typescript-client": "^0.18.0",
    "graphql-codegen-typescript-common": "^0.18.0",
    "typescript": "^3.3.3333"
  },
  "dependencies": {
    "apollo-boost": "^0.3.1",
    "graphql": "^14.1.1"
  }
}
```
`types-from-ts.ts` file

```diff
// types-from-ts.ts
export type Maybe<T> = T | null;

export enum Role {
  User = "USER",
  Admin = "ADMIN"
}

-// ====================================================
-// Documents
-// ====================================================
-
-export type FindUserVariables = {
-  userId: string;
-};
-
-export type FindUserQuery = {
-  __typename?: "Query";
-
-  user: Maybe<FindUserUser>;
-};
-
-export type FindUserUser = UserFieldsFragment;
-
-export type UserFieldsFragment = {
-  __typename?: "User";
-
-  id: string;
-
-  username: string;
-
-  role: Role;
-};
```

**Comment by dotansimha** [comment]
Hi @FuNn1esT ! 
Any chance you can share the code file you are using to store the documents? 
We are using https://github.com/DAB0mB/graphql-tag-pluck to extract the documents, and it might be a bug related to it. 
@DAB0mB Can you please take a look?

**Comment by FuNn1esT** [comment]
I've includes all files in ISSUE description.

The minimum reproduce project structure is

```
.
├── codegen.yml
├── node_modules
├── package.json
├── schema.gql
├── test.gql
├── test.ts
├── types-from-gql.ts
├── types-from-ts.ts
└── yarn.lock
```

`types-from-gql.ts` and `types-from-ts.ts` files were generated.

**Comment by shlomiatar** [comment]
We ran into the same issue. 
Diving into the graphql-tag-pluck readme, we found that the package looks for `gql` tags that were imported from graphql-tag or a few other modules, or a global `graphql` tag. 

We use our own "helper" package for importing the gql tag, so graphql-tag-pluck just skip it. 

Now I'm checking to see if there's any way to pass options from the codegen.yml to the graphql-tag-pluck so we could define the modules or the global gql tag
